src/app/elevation/qgselevationprofilewidget.cpp

```diff
@@ -1036,6 +1039,26 @@ void QgsElevationProfileWidget::createOrUpdateRubberBands( )
   }
 }
 
+void QgsElevationProfileWidget::onProjectElevationPropertiesChanged()
+{
+  // Only the vector layers whose clamping depend on the terrain need to be updated
+  // if the project elevation properties have changed.
+  // Therefore, update the plot if this criteria is met.
+  for ( QgsMapLayer *layer : mCanvas->layers() )
+  {
+    if ( layer->type() == Qgis::LayerType::Vector )
+    {
+      QgsVectorLayerElevationProperties *elevationProperties = qgis::down_cast< QgsVectorLayerElevationProperties * >( layer->elevationProperties() );
+      const Qgis::AltitudeClamping clamping = elevationProperties->clamping();
+      if ( clamping == Qgis::AltitudeClamping::Terrain || clamping == Qgis::AltitudeClamping::Relative )
```

Comment:
can you use a switch here instead (with no default clause), so that we get an explicit warning if more clamping modes are introduced in future?

Reply:
Done.